### ISSUE TITLE
FP4 block scales ignore negative outliers and clip weights

### ISSUE THREAD

**Issue by sims1253** [sims1253/starling#58]
Both `quantize_fp4` and `quantize_fp4_packed` in `src/starling/granite/fp4.py` calculate block scale using `blocks.amax(dim=-1).abs()`. That finds the largest signed value before taking its magnitude. A negative outlier larger than every positive value is therefore clipped against an undersized scale.

CPU reproduction (confirmed during cleanup):

```python
import torch
from starling.granite.fp4 import quantize_fp4, dequantize_fp4, quantize_fp4_packed

w = torch.ones(1, 16)
w[0, 0] = -6
codes, scales, meta = quantize_fp4(w)
print(scales)  # tensor([1.])
print(dequantize_fp4(codes, scales, meta)[0, 0])  # -1, instead of -6
print(quantize_fp4_packed(w)[1].float())  # tensor([[1.]])
```

Use the maximum absolute magnitude when selecting scale in both formats. Add focused coverage for negative outliers, all-negative blocks, zeros and scale underflow/overflow, then rerun quantization quality measurements before relying on previous WER or calibration conclusions. This changes model arithmetic, so it was left out of the deletion-focused cleanup.
